src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -983,16 +983,17 @@ trait Implicits {
 
       /** Sorted list of eligible implicits.
        */
-      val eligible = {
-        val matches = iss flatMap { is =>
-          val result = is filter (info => checkValid(info.sym) && survives(info))
-          shadower addInfos is
-          result
-        }
 
-        // most frequent one first
-        matches sortBy (x => if (isView) -x.useCountView else -x.useCountArg)
+      val eligible: mutable.ArrayBuffer[ImplicitInfo] = new mutable.ArrayBuffer[ImplicitInfo]()
+      iss.foreach { is =>
+        is.foreach { info =>
+          if (checkValid(info.sym) && survives(info))
+            eligible += info
+        }
+        shadower addInfos is
       }
+      eligible.sortBy(x => if (isView) -x.useCountView else -x.useCountArg)
```

Comment:
the call to `sortBy` returns a new collection that is discarded

```suggestion
      eligible.sortInPlace()(Ordering.by(x => if (isView) -x.useCountView else -x.useCountArg))
```

Reply:
Current code:
```
          // most frequent one first under Scala 2.12 mode. We've turned this optimization off to avoid
          // compilation order variation in whether a search succeeds or diverges.
```
Since the shadower will surely be retired soon, one would rather avoid tweaking, verifying, maybe breaking something.

Reply:
@NthPortal Unfortunately, this changes are done against version `2.12.x`, where the `sortInPlace` method is not yet available for the `ArrayBuffer` class. 

One possible solution to this would be to define a subclass of ArrayBuffer with such method, as an internal utility to the compiler.

Reply:
@diesalbla oh oops, I missed that